src/main/java/be/syntra/devshop/DevshopFront/services/CartServiceImpl.java

```diff
@@ -74,4 +90,41 @@ private Product getProductFromCartById(Long productId) {
                 .findFirst()
                 .orElseThrow(() -> new ProductNotFoundException("Product with id = " + productId + " was not found in your cart"));
     }
+    @Override
+    public StatusNotification payCart(String userName) {
+        setUserName(currentCart, userName);
+        log.info("username() -> {}", currentCart.getUser());
+        setCartToFinalized(currentCart);
+        HttpEntity<CartDto> requestDto = new HttpEntity<>(currentCart);
+        log.info("cart() -> {}", currentCart.toString());
+
+        ResponseEntity<CartDto> cartDtoResponseEntity = restTemplate.postForEntity(resourceUrl, requestDto, CartDto.class);
+        if (HttpStatus.CREATED.equals(cartDtoResponseEntity.getStatusCode())) {
+            log.info("payCart() -> successful {}", currentCart.getClass());
```

Comment:
why would you want to log the dto's class? You could leave the parameter out (= no {}), or just print currentCart